src/iaas-utils/ha/gcp/providers/GCPIAAS.go

```diff
@@ -0,0 +1,361 @@
+		AuthBaseURL:   iaasProvider.Config.AuthorizationBaseURL,
+		PrivateKeyId:  iaasProvider.Config.PrivateKeyId,
+		PrivateKey:    iaasProvider.Config.PrivateKey,
+		ClientEmailId: iaasProvider.Config.ClientEmailId,
+		Scopes:        strings.Split(iaasProvider.Config.Scopes, ","),
+	}
+
+	// Initialize the service client so that it can be used by other services.
+	iaasProvider.serviceClient = &clients.GCPServiceClient{}
+	returnValue = iaasProvider.serviceClient.Initialize(iaasDescriptors, authorizationRequest)
+	if returnValue != 0 {
+		// Initialization has failed and cannot proceed any further
+		return returnValue
+	}
+
+	// Initialize the services.
+	iaasProvider.vmService = &services.VMService{}
+	iaasProvider.vmService.Initialize(iaasProvider.serviceClient)
+
+	iaasProvider.vmGroupService = &services.VMGroupService{}
+	iaasProvider.vmGroupService.Initialize(iaasProvider.serviceClient)
+
+	iaasProvider.hpService = &services.HealthProbeService{}
+	iaasProvider.hpService.Initialize(iaasProvider.serviceClient)
+
+	iaasProvider.lbService = &services.LoadBalancerService{}
+	iaasProvider.lbService.Initialize(iaasProvider.serviceClient)
+
+	iaasProvider.lbRuleService = &services.LoadBalancingRuleService{}
+	iaasProvider.lbRuleService.Initialize(iaasProvider.serviceClient)
+
+	return 0
+
+}
+
+func (iaasProvider *GCPIAAS) IsHAEnabled() bool {
+	return true
+}
+
+func (iaasProvider *GCPIAAS) GetConfig() models.ConfigParams {
+	return iaasProvider.Config
+}
+
+func (iaasProvider *GCPIAAS) initializeLandscapeInfo(virtualMachine *gcpmodels.VirtualMachine) (string, string, string, string, string, string) {
+
+	var region, regionName, availabilityZone, azName, network, subNetwork string
+
+	region = virtualMachine.Region
+	temp := strings.Split(region, "/")
+	regionName = temp[(len(temp) - 1)]
+	availabilityZone = virtualMachine.Zone
+	temp = strings.Split(availabilityZone, "/")
+	azName = temp[(len(temp) - 1)]
+	log.Println("Region name is ", regionName, ". and AZ name is ", azName)
+
+	for _, currentNetworkInterface := range virtualMachine.NetworkInterfaces {
+		if currentNetworkInterface.NetworkIP == iaasProvider.Config.CurrentInstanceIP {
+			network = currentNetworkInterface.Network
+			subNetwork = currentNetworkInterface.Subnetwork
+			break
+		}
+	}
+	return region, regionName, availabilityZone, azName, network, subNetwork
+
+}
+
+func (iaasProvider *GCPIAAS) getLoadBalancerName() string {
+	return "lb-" + iaasProvider.Config.DeploymentGuid
+}
+
+func (iaasProvider *GCPIAAS) getVMGroupName() string {
+	return iaasProvider.getLoadBalancerName() + "-" + strings.Replace(iaasProvider.Config.CurrentInstanceIP, ".", "-", -1)
+}
+
+func (iaasProvider *GCPIAAS) getHealhProbeName() string {
+	return "health-check--" + iaasProvider.getLoadBalancerName()
+}
+
+func (iaasProvider *GCPIAAS) getLoadBalancingRuleName() string {
+	return "lbrule-" + iaasProvider.getLoadBalancerName()
+}
+
+func (iaasProvider *GCPIAAS) ManageResources() int {
```

Comment:
can the return values be moved to constants ? Really cryptic to understand what each return means .